truckersmp-cli

```diff
@@ -1,512 +1,547 @@
+      os.path.join(os.path.expanduser("~"), ".cache"))
+    XDG_DATA_HOME = os.getenv("XDG_DATA_HOME",
+      os.path.join(os.path.expanduser("~"), ".local/share"))
+    default_gamedir = {
+        "ats" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/American Truck Simulator/data"),
+        "ets2" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/Euro Truck Simulator 2/data"),
+    }
+    default_prefixdir = {
+        "ats" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/American Truck Simulator/prefix"),
+        "ets2" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/Euro Truck Simulator 2/prefix"),
+    }
+    default_moddir = os.path.join(XDG_DATA_HOME, "truckersmp-cli/TruckersMP")
+    default_protondir = os.path.join(XDG_DATA_HOME, "truckersmp-cli/Proton")
+    scriptdir = os.path.dirname(os.path.realpath(__file__))
+    steamcmddir = os.path.join(XDG_CACHE_HOME, "truckersmp-cli/steamcmd")
+
+
+def printdbg(msg, end = "\n"):
+    if args.verbose:
+        print(msg, end = end)
+
+def printerr(msg, end = "\n"):
+    print(msg, end = end, file = sys.stderr)
+
+def download_files(host, files_to_download):
+    try:
+        conn = http.client.HTTPSConnection(host)
+    except Exception as e:
+        printerr("Failed to connect to {}: {}".format(host, e))
+        return False
+    try:
+        while len(files_to_download) > 0:
+            path, dest, md5 = files_to_download[0]
+            md5hash = hashlib.md5()
+            bufsize = md5hash.block_size * 256
+            name = os.path.basename(path)
+            destdir = os.path.dirname(dest)
+            printdbg(
+              "Downloading file https://{}{} to {}".format(host, path, destdir))
+
+            # make file hierarchy
+            os.makedirs(destdir, exist_ok = True)
+
+            # download file
+            conn.request("GET", path, headers = {"Connection" : "keep-alive"})
+            res = conn.getresponse()
+
+            if res.status != 200:
+                printerr("Server response code is not 200.")
+                return False
+
+            lastmod = res.getheader("Last-Modified")
+            content_len = res.getheader("Content-Length")
+
+            with open(dest, "wb") as f:
+                downloaded = 0
+                while True:
+                    buf = res.read(bufsize)
+                    if not buf:
+                        break
+                    downloaded += len(buf)
+                    f.write(buf)
+                    md5hash.update(buf)
+                    if content_len:
+                        progress = "{:,} / {:,}".format(downloaded, int(content_len))
+                    else:
+                        progress = "{:,}".format(downloaded)
+                    print("\r{:40}{:>40}".format(name, progress), end = "")
+
+            if md5hash.hexdigest() != md5:
+                print("")  # newline
+                printerr("MD5 mismatch for {}".format(dest))
```

Comment:
Is it intended that the newline is written to stdout and the error message to stderr. If it's simply to let the error message breathe, send the newline to stderr too.

Reply:
Not intended to print a blank line. The `print()` for download progress doesn't print a newline:
```py
print("\r{:40}{:>40}".format(name, progress), end="")
```
Now, if MD5 doesn't match, the script prints "[filename]   MD5 ERROR" with a newline:
```py
if md5hash.hexdigest() != md5:
    print("\r{:40}{:>40}".format(name, "MD5 ERROR"))
    printerr("MD5 mismatch for {}".format(dest))
    return False
```